protocol/adapter/protocol_whisper.go

```diff
@@ -69,19 +71,14 @@ func (w *ProtocolWhisperAdapter) decodeMessage(message *whisper.ReceivedMessage)
 	payload := message.Payload
 	publicKey := message.SigToPubKey()
 	hash := message.EnvelopeHash.Bytes()
+	msg := whisper.ToWhisperMessage(message)
 
-	if w.pfs != nil {
-		decryptedPayload, err := w.pfs.HandleMessage(
-			w.transport.KeysManager().PrivateKey(),
-			publicKey,
-			payload,
-			hash,
-		)
-		if err != nil {
-			log.Printf("failed to handle message %#+x by PFS: %v", hash, err)
-		} else {
-			payload = decryptedPayload
+	if w.publisher != nil {
+		if err := w.publisher.ProcessMessage(msg, msg.Hash); err != nil {
+			// TODO(adam): err can be chat.ErrNotPairedDevice, chat.ErrDeviceNotFound and in that case it requires special handling.
```

Comment:
Published will take care of the handling, so no handling will be required (working on it now)